### ISSUE TITLE
numerous compile errors using 10.5 sdk

### ISSUE THREAD

**Issue by gghose** [jelledelaender/CMCrashReporter#1]
I was trying to compile it using the 10.5 sdk and ran into numerous problems. I couldn't find the old code anywhere. Would it possible to release the old version as a branch?

**Comment by jelledelaender** [comment]
We moved this project from SVN (Google Code) to GIT quite recently.
I'm afraid we lost all previous versions :(

10.5 is quite old, you are still building against this SDK?

**Comment by gghose** [comment]
Unfortunately, I have a framework which needs to run on PPCs, so yes I’m using the 10.5 sdk. I do all my development on an Intel running 10.10 though, which really speaks to the robustness of OS X cross-platform and cross-version development.

If you think of any magic backup locations where the old code might live, please let me know!

Geoff Ghose
Associate Professor
Departments of Neuroscience, Radiology, and Psychology
Center for Magnetic Resonance Research
University of Minnesota
http://www.ghoselab.cmrr.umn.edu
ghose@umn.edu

**Comment by jelledelaender** [comment]
I found a version of 2011 on my computer, but not sure about the state.

It can be a start, maybe you can verify with a diff what exactly changed and apply the useful changes.
Or you can copy the code to detect/read the crashes, the code to upload do a POST request and make a new interface/controller for this.

Download: http://screens.delaender.net/cmcrashreporter_2011.zip

**Comment by gghose** [comment]
Compiles perfectly, and the panel comes up offering to send the last crash report with the email filled in, but then I get the message unable to send.

Console has this:

A warning that the version is misrepresented (which really shouldn’t matter)

2015/03/24 9:51:29.675 PM AttendDynX[74050]: WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.2 instead of 10.10.2. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.
Call location:
2015/03/24 9:51:29.675 PM AttendDynX[74050]: 0   CarbonCore                          0x00007fff840c0d9b ___Gestalt_SystemVersion_block_invoke + 113
2015/03/24 9:51:29.675 PM AttendDynX[74050]: 1   libdispatch.dylib                   0x00007fff8fc67c13 _dispatch_client_callout + 8
2015/03/24 9:51:29.675 PM AttendDynX[74050]: 2   libdispatch.dylib                   0x00007fff8fc67b26 dispatch_once_f + 117
2015/03/24 9:51:29.676 PM AttendDynX[74050]: 3   CarbonCore                          0x00007fff840693ca _Gestalt_SystemVersion + 987
2015/03/24 9:51:29.676 PM AttendDynX[74050]: 4   CarbonCore                          0x00007fff84068fb7 Gestalt + 144
2015/03/24 9:51:29.676 PM AttendDynX[74050]: 5   Lablib                              0x00000001000835a6 +[CMCrashReporterGlobal isRunningLeopard] + 38
2015/03/24 9:51:29.676 PM AttendDynX[74050]: 6   Lablib                              0x0000000100118c40 +[CMCrashReporter getReports] + 112

this:

2015/03/24 9:51:29.737 PM tccd[55827]: Prompting for access to kTCCServiceAddressBook from /Users/geoff/Desktop/Lablib/AttendDynX5/build/Development/AttendDynX.app/Contents/MacOS/AttendDynX

and this:

2015/03/24 9:52:06.056 PM AttendDynX[74050]: Warning: No CMSubmitURL - key available for CMCrashReporter. Please add this key at your info.plist file.

Not sure what’s up.

**Comment by jelledelaender** [comment]
@gghose Did you read the README? See https://github.com/CodingMammoth/CMCrashReporter/blob/master/README.md

There is a point about the CMSubmitURL which you are clearly missing.

The other warnings are system warnings that a method is used that shouldn't be used anymore. You should change this, the log is even saying which other class you should use. This is probability already fixed in the latest version... so you can take a look at the current master and compare.

**Comment by gghose** [comment]
Yes, I did define CMSubmitURL in a plist within a framework….. perhaps its not be accessed by the app, I’ll look into it. But my understanding was that it’s simply a http posting setting, and that, even if it wasn’t defined, the emailing functionality should still work.

**Comment by jelledelaender** [comment]
The code/framework itself wont send the email. To archive that, you need to implement the SMTP-protocol, or use a SMTP framework. Knowing you need to SMTP credentials of the device of the user, or be using a global server and including your credentials in the source-code wouldn't be a good idea.

CMCrashReporter is only doing a HTTP Post request to a server. This server can then easily send the report via email, or store it in a DB, or ...

You should check where it's using CMSubmitURL, I think it's using atm the mainBundle, so it should be in the plist of the application. You can change it to the bundle of the framework, then defining it in the plist of the framework should work.

**Comment by gghose** [comment]
Gotcha….. looking at the URL result, it seems like I’ve got a web server configuration problem; I’ll have to talk to the administrator.

In the meantime, the fix for looking up CMSubmitURL when using frameworks is simple

in crashReportURL

replace

value = [[[NSBundle mainBundle] infoDictionary]

with

value = [[[NSBundle bundleForClass:[self class]] infoDictionary]

Thanks for your help.
